common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -16,6 +16,61 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     document
   }
 
+  /////////////////////////////
+  // External video cleaner
+  /////////////////////////////
+
+  "AmpEmbedCleaner" should "replace an iframe in a YouTube video-element with an amp-youtube element" in {
+    val youtubeUrl = "https://www.youtube.com/watch?v=foo_12-34"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$youtubeUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-youtube").size should be(1)
+  }
+
+  "AmpEmbedCleaner" should "not create an amp-youtube element if videoid missing" in {
+    val youtubeUrl = s"http://www.youtube.com/"
```

Comment:
It doesn't look like it matters. Even if the canonical URL is `http`, `amp-youtube` and `amp-vimeo` still request the `https` version of the video.

![picture 123](https://cloud.githubusercontent.com/assets/5931528/19923493/8df0189c-a0df-11e6-9eb0-0b64b7b3d39c.png)